pynucastro/eos/helmholtz.py

```diff
@@ -0,0 +1,743 @@
+
+        def _extract_line_data(line):
+            line_entries = line.strip().split()
+            f = np.array([np.float64(s) for s in line_entries])
+            return f
+
+        with open(self.table, 'r') as file:
+
+            f_local = np.empty((self.imax, self.jmax, 9))
+            for j in range(0,self.jmax):
+                for i in range(0,self.imax):
+                    line = file.readline()
+                    f_local[i,j,:] = _extract_line_data(line)[:]
+
+            pd_local = np.empty((self.imax, self.jmax, 4))
+            for j in range(0,self.jmax):
+                for i in range(0,self.imax):
+                    line = file.readline()
+                    pd_local[i,j,:] = _extract_line_data(line)[:]
+
+            ef_local = np.empty((self.imax, self.jmax, 4))
+            for j in range(0,self.jmax):
+                for i in range(0,self.imax):
+                    line = file.readline()
+                    ef_local[i,j,:] = _extract_line_data(line)[:]
+
+            xnf_local = np.empty((self.imax, self.jmax, 4))
+            for j in range(0,self.jmax):
+                for i in range(0,self.imax):
+                    line = file.readline()
+                    xnf_local[i,j, :] = _extract_line_data(line)[:]
+
+        self.f = f_local
+        self.pd = pd_local
+        self.ef = ef_local
+        self.xnf = xnf_local
+
+    #Let us define the bicubic polynomials:
+    #======================================
+
+    # xpsi0
+    def _xpsi0(self, z):
+
+        return z * z * (2.0 * z - 3.0) + 1.0
+
+    def _xdpsi0(self, z):
+
+        return z * (6.0 * z - 6.0)
+
+    # xpsi1
+    def _xpsi1(self, z):
+
+        return z * (z * (z - 2.0) + 1.0)
+
+    def _xdpsi1(self, z):
+
+        return z * (3.0 * z - 4.0) + 1.0
+
+    #Let us define the biquintic the polynomials:
+    #============================================
+
+    # psi0
+    def _psi0(self, z):
+
+        return  z * z * z * (z * (-6.0 * z + 15.0) -10.0) + 1.0
+
+    def _dpsi0(self,z):
+
+        return z * z * (z * (-30.0*z + 60.0) - 30.0)
+
+    def _ddpsi0(self,z):
+
+        return z * (z * (-120.0 * z + 180.0) - 60.0)
+
+    # psi1
+    def _psi1(self, z):
+
+        return z * ( z * z * (z * (-3.0 * z + 8.0) - 6.0) + 1.0)
+
+    def _dpsi1(self,z):
+
+        return z * z * ( z * (-15.0 * z + 32.0) - 18.0) + 1.0
+
+    def _ddpsi1(self,z):
+
+        return z * (z * (-60.0 * z + 96.0) - 36.0)
+
+    # psi2
+    def _psi2(self, z):
+
+        return 0.5 * z * z * ( z * ( z * (-z + 3.0) - 3.0) + 1.0)
+
+    def _dpsi2(self,z):
+
+        return 0.5 * z * (z * (z * (-5.0 * z + 12.0) - 9.0) + 2.0)
+
+    def _ddpsi2(self,z):
+
+        return 0.5 * (z * ( z * (-20.0 * z + 36.0) - 18.0) + 2.0)
+
+    # Let us read the density coeficients that participate in the
+    # interpolation:
+    #============================================================
+
+    def _read_wd(self, der=0, sel='biquintic'):
```

Comment:
`_read_wd()` and `_read_wt()` are almost identical: most of the code should be pulled out into a shared helper function